c/collections.c

```diff
@@ -1237,7 +1234,17 @@ void qInsert(Queue *q, void *newData){
   return;
 }
 
-void *qRemove(Queue *q){
+void qInsert(Queue *q, void *newData) {
```

Comment:
newData should be const

Reply:
I'd need to think this through. If we make this const, we need to make the field const and `qRemove` will need to return a const value which will break existing code. I don't see how we can solve this without casting at some point.

Reply:
I think a cast is fine. I would  cast to (void *) when assigning the data element and leave everything else as is. `qRemove` should remain as is. That's pretty standard. Making newData const signifies the intent of the function in that it doesn't mutate newData.